engines/prover.cpp

```diff
@@ -50,8 +52,14 @@ void Prover::initialize()
   }
 
   reached_k_ = -1;
-  bad_ = solver_->make_term(smt::PrimOp::Not, property_.prop());
-  assert(ts_->only_curr(bad_));
+
+  if (!bad_) {
+    // initialize bad_ if it is not set already
+    const Term &prop_term = (ts_.solver() == orig_property_.solver())
+      ? orig_property_.prop()
+      : to_prover_solver_.transfer_term(orig_property_.prop());
+    bad_ = solver_->make_term(smt::PrimOp::Not, prop_term);
```

Comment:
Could we still have a check here that `bad_` only contains current state variables?

Reply:
done in c03dd5d